src/Arcus.Security.Core/ISecretProvider.cs

```diff
@@ -30,3 +31,212 @@ public interface ISecretProvider
         Task<Secret> GetSecretAsync(string secretName);
     }
 }
+
+namespace Arcus.Security
+{
+    /// <summary>
+    /// Represents a provider that can retrieve secrets based on a given name.
+    /// </summary>
+    internal interface ISecretProvider
+    {
+        /// <summary>
+        /// Gets the secret by its name from the registered provider.
+        /// </summary>
+        /// <param name="secretName">The name to identity the stored secret.</param>
+        /// <returns>
+        ///     <para>[Success] when the secret with the provided <paramref name="secretName"/> was found;</para>
+        ///     <para>[Failure] when the secret could not be retrieved via the provider.</para>
+        /// </returns>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="secretName"/> is blank.</exception>
+        Task<SecretResult> GetSecretAsync(string secretName)
+        {
+            return Task.FromResult(GetSecret(secretName));
```

Comment:
Aren't we going to shoot ourselves in the foot with this default interface method ?
It looks like the 'Async' version for retrieving a secret won't be really 'async'.

Reply:
Currently, a lot of our secret providers are solely 'sync' providers without an 'async' possibility: environment variables, configuration, Docker Secrets, User Secrets, Command Line... are all solely 'sync'. Having this in the interface greatly reduces the noise when implementing it (which would mean moving/duplicating the selected line to each and every one of those secret providers ➡️ which is currently done).